{
 "cells": [
  {
   "cell_type": "raw",
   "id": "d1406aa9-d79b-45dc-895d-a683dfa31dcb",
   "metadata": {},
   "source": [
    "---\n",
    "title: Get sequence of variants 01\n",
    "subtitle: Set genomic interval and get sequence for observed/unobserved alleles\n",
    "format:\n",
    "    html:\n",
    "        code-fold:  true\n",
    "        code-tools: true\n",
    "---"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "a7385375-bebe-45e7-bd36-7f8532bd7394",
   "metadata": {},
   "source": [
    "**Set environment**"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "4aeceab6-272e-4a1c-82b3-d425840ea783",
   "metadata": {},
   "outputs": [],
   "source": [
    "import numpy  as np\n",
    "import pandas as pd\n",
    "import itertools as it\n",
    "from functools import partial\n",
    "import os, sys, re\n",
    "import csv"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "936e6a82-85ce-4ce9-a158-88639777b9b5",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "BASE DIRECTORY (FD_BASE): /hpc/group/igvf/kk319\n",
      "REPO DIRECTORY (FD_REPO): /hpc/group/igvf/kk319/repo\n",
      "WORK DIRECTORY (FD_WORK): /hpc/group/igvf/kk319/work\n",
      "DATA DIRECTORY (FD_DATA): /hpc/group/igvf/kk319/data\n",
      "\n",
      "\n",
      "You are working with      IGVF BlueSTARR\n",
      "PATH OF PROJECT (FD_PRJ): /hpc/group/igvf/kk319/repo/Proj_IGVF_BlueSTARR\n",
      "PROJECT RESULTS (FD_RES): /hpc/group/igvf/kk319/repo/Proj_IGVF_BlueSTARR/results\n",
      "PROJECT SCRIPTS (FD_EXE): /hpc/group/igvf/kk319/repo/Proj_IGVF_BlueSTARR/scripts\n",
      "PROJECT DATA    (FD_DAT): /hpc/group/igvf/kk319/repo/Proj_IGVF_BlueSTARR/data\n",
      "PROJECT NOTE    (FD_NBK): /hpc/group/igvf/kk319/repo/Proj_IGVF_BlueSTARR/notebooks\n",
      "PROJECT DOCS    (FD_DOC): /hpc/group/igvf/kk319/repo/Proj_IGVF_BlueSTARR/docs\n",
      "PROJECT LOG     (FD_LOG): /hpc/group/igvf/kk319/repo/Proj_IGVF_BlueSTARR/log\n",
      "PROJECT REF     (FD_REF): /hpc/group/igvf/kk319/repo/Proj_IGVF_BlueSTARR/references\n",
      "\n"
     ]
    }
   ],
   "source": [
    "%run ../run_config_project.py\n",
    "show_env()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "7c544f35-4dd0-40a6-80b6-40c27551cd48",
   "metadata": {},
   "outputs": [],
   "source": [
    "FP_GEN = \"/hpc/group/igvf/kk319/data/genome/hg38/hg38.fa\"\n",
    "\n",
    "import pysam\n",
    "fasta  = pysam.FastaFile(FP_GEN)\n",
    "\n",
    "#from pyfaidx import Fasta\n",
    "#fasta  = Fasta(FP_GEN)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "7ea68b38-a9a1-41da-9a06-6f6589ef63fd",
   "metadata": {},
   "source": [
    "## Import data"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "ae5caa0b-1fd9-49bb-a450-b932320fc9f5",
   "metadata": {},
   "source": [
    "**Check file existent**"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "0f2d370c-f83e-4355-a2da-bd98e9f113b7",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/hpc/group/igvf/kk319/repo/Proj_IGVF_BlueSTARR/data\n"
     ]
    }
   ],
   "source": [
    "%%bash -s \"{FD_DAT}\"\n",
    "echo $1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "357e4d80-51ec-4870-8b18-6348521d737b",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "annotated_snv_sites_filtered.tsv.gz  regions.fasta.gz\n",
      "annotated_snv_sites.tsv.gz\t     regions_GoF_only.fasta.gz\n"
     ]
    }
   ],
   "source": [
    "!ls $FD_DAT/variant_bluestarr_richard/closed"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "4c98bf11-11e4-4481-8a74-4a10fb233e59",
   "metadata": {},
   "source": [
    "**Import table**"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "e6bc2e7f-707a-4d3e-98e1-4e70182092c4",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(1884977, 6)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>region</th>\n",
       "      <th>pos0</th>\n",
       "      <th>delta</th>\n",
       "      <th>ref</th>\n",
       "      <th>obs</th>\n",
       "      <th>unobs</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr4:74487576-74488141</td>\n",
       "      <td>74487586</td>\n",
       "      <td>0.964984</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr9:135685425-135685942</td>\n",
       "      <td>135685432</td>\n",
       "      <td>0.945423</td>\n",
       "      <td>G</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1:173251481-173251541</td>\n",
       "      <td>173251490</td>\n",
       "      <td>0.910127</td>\n",
       "      <td>G</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1:147523084-147523503</td>\n",
       "      <td>147523391</td>\n",
       "      <td>0.908800</td>\n",
       "      <td>G</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr18:54580694-54581420</td>\n",
       "      <td>54580749</td>\n",
       "      <td>0.862296</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                     region       pos0     delta ref obs unobs\n",
       "0    chr4:74487576-74488141   74487586  0.964984   A   G     C\n",
       "1  chr9:135685425-135685942  135685432  0.945423   G   G     C\n",
       "2  chr1:173251481-173251541  173251490  0.910127   G   G     C\n",
       "3  chr1:147523084-147523503  147523391  0.908800   G   G     C\n",
       "4   chr18:54580694-54581420   54580749  0.862296   A   G     T"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "### set file directory\n",
    "txt_fdiry = os.path.join(FD_DAT, \"variant_bluestarr_richard\", \"closed\")\n",
    "txt_fname = \"annotated_snv_sites_filtered.tsv.gz\"\n",
    "txt_fpath = os.path.join(txt_fdiry, txt_fname)\n",
    "\n",
    "### read table\n",
    "dat = pd.read_csv(txt_fpath, sep = \"\\t\")\n",
    "\n",
    "### assign and show\n",
    "dat_variant_import = dat\n",
    "print(dat.shape)\n",
    "dat.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "6410223e-45ec-4f48-8158-69d596f1c792",
   "metadata": {},
   "source": [
    "## Arrange table"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "b5063286-2ad9-4ef5-b8c5-dbe47cc78690",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(1884977, 10)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chrom</th>\n",
       "      <th>ChromStart</th>\n",
       "      <th>ChromEnd</th>\n",
       "      <th>Region</th>\n",
       "      <th>Variant_ID</th>\n",
       "      <th>Pos0</th>\n",
       "      <th>Ref</th>\n",
       "      <th>Obs</th>\n",
       "      <th>Unobs</th>\n",
       "      <th>Delta</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr4</td>\n",
       "      <td>74487576</td>\n",
       "      <td>74488141</td>\n",
       "      <td>chr4:74487576-74488141</td>\n",
       "      <td>chr4:74487586:A:G:C</td>\n",
       "      <td>74487586</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>0.964984</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr9</td>\n",
       "      <td>135685425</td>\n",
       "      <td>135685942</td>\n",
       "      <td>chr9:135685425-135685942</td>\n",
       "      <td>chr9:135685432:G:G:C</td>\n",
       "      <td>135685432</td>\n",
       "      <td>G</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>0.945423</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>173251481</td>\n",
       "      <td>173251541</td>\n",
       "      <td>chr1:173251481-173251541</td>\n",
       "      <td>chr1:173251490:G:G:C</td>\n",
       "      <td>173251490</td>\n",
       "      <td>G</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>0.910127</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>147523084</td>\n",
       "      <td>147523503</td>\n",
       "      <td>chr1:147523084-147523503</td>\n",
       "      <td>chr1:147523391:G:G:C</td>\n",
       "      <td>147523391</td>\n",
       "      <td>G</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>0.908800</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr18</td>\n",
       "      <td>54580694</td>\n",
       "      <td>54581420</td>\n",
       "      <td>chr18:54580694-54581420</td>\n",
       "      <td>chr18:54580749:A:G:T</td>\n",
       "      <td>54580749</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>0.862296</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   Chrom  ChromStart   ChromEnd                    Region  \\\n",
       "0   chr4    74487576   74488141    chr4:74487576-74488141   \n",
       "1   chr9   135685425  135685942  chr9:135685425-135685942   \n",
       "2   chr1   173251481  173251541  chr1:173251481-173251541   \n",
       "3   chr1   147523084  147523503  chr1:147523084-147523503   \n",
       "4  chr18    54580694   54581420   chr18:54580694-54581420   \n",
       "\n",
       "             Variant_ID       Pos0 Ref Obs Unobs     Delta  \n",
       "0   chr4:74487586:A:G:C   74487586   A   G     C  0.964984  \n",
       "1  chr9:135685432:G:G:C  135685432   G   G     C  0.945423  \n",
       "2  chr1:173251490:G:G:C  173251490   G   G     C  0.910127  \n",
       "3  chr1:147523391:G:G:C  147523391   G   G     C  0.908800  \n",
       "4  chr18:54580749:A:G:T   54580749   A   G     T  0.862296  "
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "### init\n",
    "dat = dat_variant_import.copy()\n",
    "\n",
    "### split region into chrom/start/end\n",
    "dat_chrom_info    = dat[\"region\"].str.split(\"[:-]\", expand=True)\n",
    "dat[\"Chrom\"]      = dat_chrom_info[0]\n",
    "dat[\"ChromStart\"] = dat_chrom_info[1].astype(int)\n",
    "dat[\"ChromEnd\"]   = dat_chrom_info[2].astype(int)\n",
    "\n",
    "### rename original columns\n",
    "dat = dat.rename(columns={\n",
    "    \"region\": \"Region\",\n",
    "    \"pos0\":   \"Pos0\",\n",
    "    \"delta\":  \"Delta\",\n",
    "    \"ref\":    \"Ref\",\n",
    "    \"obs\":    \"Obs\",\n",
    "    \"unobs\":  \"Unobs\"\n",
    "})\n",
    "\n",
    "### add Variant_ID\n",
    "dat[\"Variant_ID\"] = (\n",
    "    dat[\"Chrom\"].astype(str) + \":\" +\n",
    "    dat[\"Pos0\"].astype(str)  + \":\" +\n",
    "    dat[\"Ref\"].astype(str)   + \":\" +\n",
    "    dat[\"Obs\"].astype(str)   + \":\" +\n",
    "    dat[\"Unobs\"].astype(str)\n",
    ")\n",
    "\n",
    "### Reorder columns\n",
    "dat = dat[[\n",
    "    \"Chrom\", \"ChromStart\", \"ChromEnd\", \"Region\", \n",
    "    \"Variant_ID\", \n",
    "    \"Pos0\",\n",
    "    \"Ref\", \"Obs\", \"Unobs\",\n",
    "    \"Delta\"\n",
    "]]\n",
    "\n",
    "### assign and show\n",
    "dat_variant_arrange = dat\n",
    "print(dat.shape)\n",
    "dat.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "618c9bdd-cc37-4820-9956-6a2d811b9312",
   "metadata": {},
   "source": [
    "**Export table**"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "0e736466-087b-4be5-8bd0-288dfbdc4d09",
   "metadata": {},
   "outputs": [],
   "source": [
    "### set file directory\n",
    "txt_fdiry = os.path.join(FD_RES, \"analysis_variant_motif_richard\")\n",
    "txt_fname = \"variant_closed_gof_bluestarr.tsv.gz\"\n",
    "txt_fpath = os.path.join(txt_fdiry, txt_fname)\n",
    "\n",
    "### write table\n",
    "dat_variant_arrange.to_csv(txt_fpath, sep=\"\\t\", index=False, compression=\"gzip\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "1cf3332d-a5df-449f-87ac-2374a84e965d",
   "metadata": {},
   "source": [
    "## Sanity check: reference allele"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "ab4da9bb-939c-4de7-b53c-e64c2224519d",
   "metadata": {},
   "source": [
    "**Check the first row**"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "f4f87d2e-841b-483d-9493-d15649330a04",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Chrom: chr4; Start-End: 74487576-74488141; Pos: 74487586\n"
     ]
    }
   ],
   "source": [
    "txt_region = \"chr4:74487576-74488141\"\n",
    "\n",
    "#txt_chrom_name, txt_chrom_start, txt_chrom_end = re.split(':|-', txt_region)\n",
    "txt_chrom_name, txt_chrom_slice = txt_region.split(\":\")\n",
    "txt_chrom_name  = str(txt_chrom_name)\n",
    "txt_chrom_pos   = 74487586\n",
    "\n",
    "print(f\"Chrom: {txt_chrom_name}; Start-End: {txt_chrom_slice}; Pos: {txt_chrom_pos}\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "51e2e29a-efbb-4c92-a5e0-b508635a185b",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "A\n"
     ]
    }
   ],
   "source": [
    "### # 0-based slicing\n",
    "print(fasta.fetch(txt_chrom_name, txt_chrom_pos, txt_chrom_pos+1))\n",
    "#print(fasta[txt_chrom_name][txt_chrom_pos:txt_chrom_pos+1].seq)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "2f25324a-ed71-4d06-9042-3a51e5d74c07",
   "metadata": {},
   "source": [
    "**Check more positions**"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "edca653d-acf1-485a-ba27-b9c3ba7aa150",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Checked 10000 variants\n",
      "Found 0 mismatches\n"
     ]
    }
   ],
   "source": [
    "### init\n",
    "dat = dat_variant_import\n",
    "dat = dat.sample(10000, random_state=1)\n",
    "\n",
    "### fetch ref allele and check mismatches\n",
    "lst_mismatches = []\n",
    "\n",
    "for txt_chrom_name, num_chrom_pos, txt_allele_ref_table in zip(\n",
    "    dat[\"region\"].str.split(\":\").str[0],\n",
    "    dat[\"pos0\"], \n",
    "    dat[\"ref\"]\n",
    "):\n",
    "    ### query the reference allele\n",
    "    ### note: pos is 0-based, end-exclusive\n",
    "    txt_allele_ref_fasta = fasta.fetch(txt_chrom_name, num_chrom_pos, num_chrom_pos+1) \n",
    "\n",
    "    ### convert to uppercase for simplicity\n",
    "    ### uppercase: high-confidence sequence\n",
    "    ### lowercase: soft-masked sequence\n",
    "    txt_allele_ref_fasta = txt_allele_ref_fasta.upper()\n",
    "    \n",
    "    ### check match/mismatch\n",
    "    if txt_allele_ref_fasta != txt_allele_ref_table:\n",
    "        tmp = (txt_chrom_name, num_chrom_pos, txt_allele_ref_table, txt_allele_ref_fasta)\n",
    "        lst_mismatches.append(tmp)\n",
    "\n",
    "print(f\"Checked {len(dat)} variants\")\n",
    "print(f\"Found {len(lst_mismatches)} mismatches\")\n",
    "if lst_mismatches:\n",
    "    print(\"Example mismatches:\", lst_mismatches[:5])"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "0067d1d9-de4e-4153-b287-1737742382d4",
   "metadata": {},
   "source": [
    "## Helper function"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "e5585a0d-f725-4ad6-a597-a878e12392c5",
   "metadata": {},
   "outputs": [],
   "source": [
    "def get_variant_interval_series(row, num_interval_flank=35):\n",
    "    txt_chrom_name  = row[\"Region\"].split(\":\")[0]\n",
    "    txt_chrom_pos0  = row[\"Pos0\"]\n",
    "    txt_allele_ref  = row[\"Ref\"]\n",
    "    txt_allele_obs  = row[\"Obs\"]\n",
    "    txt_allele_unobs= row[\"Unobs\"]\n",
    "\n",
    "    # fetch genomic interval centered at variant position\n",
    "    txt_seq_ref = fasta.fetch(\n",
    "        txt_chrom_name,\n",
    "        txt_chrom_pos0 - num_interval_flank,\n",
    "        txt_chrom_pos0 + num_interval_flank + 1\n",
    "    ).upper()\n",
    "\n",
    "    # sanity check reference allele\n",
    "    if txt_seq_ref[num_interval_flank].upper() != txt_allele_ref.upper():\n",
    "        raise ValueError(\n",
    "            f\"Reference mismatch at {txt_chrom_name}:{txt_chrom_pos0} \"\n",
    "            f\"(expected {txt_allele_ref}, got {txt_seq_ref[num_interval_flank]})\"\n",
    "        )\n",
    "\n",
    "    # build alt windows\n",
    "    txt_seq_obs   = txt_seq_ref[:num_interval_flank] + txt_allele_obs   + txt_seq_ref[num_interval_flank+1:]\n",
    "    txt_seq_unobs = txt_seq_ref[:num_interval_flank] + txt_allele_unobs + txt_seq_ref[num_interval_flank+1:]\n",
    "\n",
    "    # flank-dependent columns only\n",
    "    return pd.Series({\n",
    "        \"Flank\":      num_interval_flank,\n",
    "        \"Interval\":   f\"{txt_chrom_name}:{txt_chrom_pos0 - num_interval_flank}-{txt_chrom_pos0 + num_interval_flank + 1}\",\n",
    "        \"Seq_Length\": len(txt_seq_ref),\n",
    "        \"Seq_Ref\":    txt_seq_ref,\n",
    "        \"Seq_Obs\":    txt_seq_obs,\n",
    "        \"Seq_Unobs\":  txt_seq_unobs,\n",
    "        \"Check_RefMatch\": (txt_seq_ref[num_interval_flank].upper() == txt_allele_ref.upper())\n",
    "    })\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "69f074d2-4420-43fc-bed2-6f9fbf08bd6e",
   "metadata": {},
   "source": [
    "## Get sequence by specifying flanking size as 35bp"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "a823ea40-7936-4f36-9fb3-58e60e958738",
   "metadata": {},
   "outputs": [],
   "source": [
    "### spanning x bp at the center of each variant\n",
    "### here the flanking region is set as 35 bp at both side\n",
    "NUM_INTERVAL_FLANK = 35 "
   ]
  },
  {
   "cell_type": "markdown",
   "id": "0cce7c4d-671f-498e-b66a-0fda0ac600b9",
   "metadata": {},
   "source": [
    "**Test run**"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "92983ee5-bd3d-4466-a029-1323762811e3",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(10000, 17)\n",
      "CPU times: user 1.54 s, sys: 225 ms, total: 1.77 s\n",
      "Wall time: 1.77 s\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chrom</th>\n",
       "      <th>ChromStart</th>\n",
       "      <th>ChromEnd</th>\n",
       "      <th>Region</th>\n",
       "      <th>Variant_ID</th>\n",
       "      <th>Pos0</th>\n",
       "      <th>Ref</th>\n",
       "      <th>Obs</th>\n",
       "      <th>Unobs</th>\n",
       "      <th>Delta</th>\n",
       "      <th>Flank</th>\n",
       "      <th>Interval</th>\n",
       "      <th>Seq_Length</th>\n",
       "      <th>Seq_Ref</th>\n",
       "      <th>Seq_Obs</th>\n",
       "      <th>Seq_Unobs</th>\n",
       "      <th>Check_RefMatch</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1467403</th>\n",
       "      <td>chr2</td>\n",
       "      <td>181868375</td>\n",
       "      <td>181869091</td>\n",
       "      <td>chr2:181868375-181869091</td>\n",
       "      <td>chr2:181869026:G:A:T</td>\n",
       "      <td>181869026</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>0.001980</td>\n",
       "      <td>35</td>\n",
       "      <td>chr2:181868991-181869062</td>\n",
       "      <td>71</td>\n",
       "      <td>CTTGATCCTCATAATCCTCACAATAACTCTAGAGAGTGAGAAGTGT...</td>\n",
       "      <td>CTTGATCCTCATAATCCTCACAATAACTCTAGAGAATGAGAAGTGT...</td>\n",
       "      <td>CTTGATCCTCATAATCCTCACAATAACTCTAGAGATTGAGAAGTGT...</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>176754</th>\n",
       "      <td>chr1</td>\n",
       "      <td>159747007</td>\n",
       "      <td>159747231</td>\n",
       "      <td>chr1:159747007-159747231</td>\n",
       "      <td>chr1:159747115:T:T:C</td>\n",
       "      <td>159747115</td>\n",
       "      <td>T</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.018575</td>\n",
       "      <td>35</td>\n",
       "      <td>chr1:159747080-159747151</td>\n",
       "      <td>71</td>\n",
       "      <td>CGTGCATTTTTGTGACCCCAATCATTTTTGAAAACTATCTCAGAGC...</td>\n",
       "      <td>CGTGCATTTTTGTGACCCCAATCATTTTTGAAAACTATCTCAGAGC...</td>\n",
       "      <td>CGTGCATTTTTGTGACCCCAATCATTTTTGAAAACCATCTCAGAGC...</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>227022</th>\n",
       "      <td>chr11</td>\n",
       "      <td>55372676</td>\n",
       "      <td>55374644</td>\n",
       "      <td>chr11:55372676-55374644</td>\n",
       "      <td>chr11:55374072:T:A:C</td>\n",
       "      <td>55374072</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>0.016494</td>\n",
       "      <td>35</td>\n",
       "      <td>chr11:55374037-55374108</td>\n",
       "      <td>71</td>\n",
       "      <td>AGCCACCTCCTAAAATATATTTGTATACATACTATTTCTATTTTAT...</td>\n",
       "      <td>AGCCACCTCCTAAAATATATTTGTATACATACTATATCTATTTTAT...</td>\n",
       "      <td>AGCCACCTCCTAAAATATATTTGTATACATACTATCTCTATTTTAT...</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>471983</th>\n",
       "      <td>chr7</td>\n",
       "      <td>73751160</td>\n",
       "      <td>73751171</td>\n",
       "      <td>chr7:73751160-73751171</td>\n",
       "      <td>chr7:73751165:G:G:T</td>\n",
       "      <td>73751165</td>\n",
       "      <td>G</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>0.010665</td>\n",
       "      <td>35</td>\n",
       "      <td>chr7:73751130-73751201</td>\n",
       "      <td>71</td>\n",
       "      <td>TAAGTTTCAAGCTTTAAGACCAAAAGGGTTGACCCGCACCATGGCT...</td>\n",
       "      <td>TAAGTTTCAAGCTTTAAGACCAAAAGGGTTGACCCGCACCATGGCT...</td>\n",
       "      <td>TAAGTTTCAAGCTTTAAGACCAAAAGGGTTGACCCTCACCATGGCT...</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1118365</th>\n",
       "      <td>chr8</td>\n",
       "      <td>72987671</td>\n",
       "      <td>72988177</td>\n",
       "      <td>chr8:72987671-72988177</td>\n",
       "      <td>chr8:72987743:G:G:T</td>\n",
       "      <td>72987743</td>\n",
       "      <td>G</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>0.004071</td>\n",
       "      <td>35</td>\n",
       "      <td>chr8:72987708-72987779</td>\n",
       "      <td>71</td>\n",
       "      <td>TTAATTAAAGATTACAAACAAAGCTGAAAACCTAGGAGAAGATAGG...</td>\n",
       "      <td>TTAATTAAAGATTACAAACAAAGCTGAAAACCTAGGAGAAGATAGG...</td>\n",
       "      <td>TTAATTAAAGATTACAAACAAAGCTGAAAACCTAGTAGAAGATAGG...</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "         Chrom  ChromStart   ChromEnd                    Region  \\\n",
       "1467403   chr2   181868375  181869091  chr2:181868375-181869091   \n",
       "176754    chr1   159747007  159747231  chr1:159747007-159747231   \n",
       "227022   chr11    55372676   55374644   chr11:55372676-55374644   \n",
       "471983    chr7    73751160   73751171    chr7:73751160-73751171   \n",
       "1118365   chr8    72987671   72988177    chr8:72987671-72988177   \n",
       "\n",
       "                   Variant_ID       Pos0 Ref Obs Unobs     Delta  Flank  \\\n",
       "1467403  chr2:181869026:G:A:T  181869026   G   A     T  0.001980     35   \n",
       "176754   chr1:159747115:T:T:C  159747115   T   T     C  0.018575     35   \n",
       "227022   chr11:55374072:T:A:C   55374072   T   A     C  0.016494     35   \n",
       "471983    chr7:73751165:G:G:T   73751165   G   G     T  0.010665     35   \n",
       "1118365   chr8:72987743:G:G:T   72987743   G   G     T  0.004071     35   \n",
       "\n",
       "                         Interval  Seq_Length  \\\n",
       "1467403  chr2:181868991-181869062          71   \n",
       "176754   chr1:159747080-159747151          71   \n",
       "227022    chr11:55374037-55374108          71   \n",
       "471983     chr7:73751130-73751201          71   \n",
       "1118365    chr8:72987708-72987779          71   \n",
       "\n",
       "                                                   Seq_Ref  \\\n",
       "1467403  CTTGATCCTCATAATCCTCACAATAACTCTAGAGAGTGAGAAGTGT...   \n",
       "176754   CGTGCATTTTTGTGACCCCAATCATTTTTGAAAACTATCTCAGAGC...   \n",
       "227022   AGCCACCTCCTAAAATATATTTGTATACATACTATTTCTATTTTAT...   \n",
       "471983   TAAGTTTCAAGCTTTAAGACCAAAAGGGTTGACCCGCACCATGGCT...   \n",
       "1118365  TTAATTAAAGATTACAAACAAAGCTGAAAACCTAGGAGAAGATAGG...   \n",
       "\n",
       "                                                   Seq_Obs  \\\n",
       "1467403  CTTGATCCTCATAATCCTCACAATAACTCTAGAGAATGAGAAGTGT...   \n",
       "176754   CGTGCATTTTTGTGACCCCAATCATTTTTGAAAACTATCTCAGAGC...   \n",
       "227022   AGCCACCTCCTAAAATATATTTGTATACATACTATATCTATTTTAT...   \n",
       "471983   TAAGTTTCAAGCTTTAAGACCAAAAGGGTTGACCCGCACCATGGCT...   \n",
       "1118365  TTAATTAAAGATTACAAACAAAGCTGAAAACCTAGGAGAAGATAGG...   \n",
       "\n",
       "                                                 Seq_Unobs  Check_RefMatch  \n",
       "1467403  CTTGATCCTCATAATCCTCACAATAACTCTAGAGATTGAGAAGTGT...            True  \n",
       "176754   CGTGCATTTTTGTGACCCCAATCATTTTTGAAAACCATCTCAGAGC...            True  \n",
       "227022   AGCCACCTCCTAAAATATATTTGTATACATACTATCTCTATTTTAT...            True  \n",
       "471983   TAAGTTTCAAGCTTTAAGACCAAAAGGGTTGACCCTCACCATGGCT...            True  \n",
       "1118365  TTAATTAAAGATTACAAACAAAGCTGAAAACCTAGTAGAAGATAGG...            True  "
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "%%time\n",
    "\n",
    "### init\n",
    "dat = dat_variant_arrange.copy()\n",
    "dat = dat.sample(10000, random_state=123)\n",
    "\n",
    "### add sequences\n",
    "fun = partial(get_variant_interval_series, num_interval_flank=NUM_INTERVAL_FLANK)\n",
    "lst = [\"Flank\", \"Interval\", \"Seq_Length\", \"Seq_Ref\", \"Seq_Obs\", \"Seq_Unobs\", \"Check_RefMatch\"]\n",
    "dat[lst] = dat.apply(fun, axis=1)\n",
    "\n",
    "### assign and show\n",
    "dat_variant_withseq_sample = dat\n",
    "print(dat.shape)\n",
    "dat.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "8d32afdc-c7fb-462d-bf66-51ecfa338662",
   "metadata": {},
   "source": [
    "**Get sequence for full table**"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "28ef5599-d4cf-4186-9a1b-6149df045498",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(1884977, 17)\n",
      "CPU times: user 3min 51s, sys: 26.3 s, total: 4min 17s\n",
      "Wall time: 4min 18s\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chrom</th>\n",
       "      <th>ChromStart</th>\n",
       "      <th>ChromEnd</th>\n",
       "      <th>Region</th>\n",
       "      <th>Variant_ID</th>\n",
       "      <th>Pos0</th>\n",
       "      <th>Ref</th>\n",
       "      <th>Obs</th>\n",
       "      <th>Unobs</th>\n",
       "      <th>Delta</th>\n",
       "      <th>Flank</th>\n",
       "      <th>Interval</th>\n",
       "      <th>Seq_Length</th>\n",
       "      <th>Seq_Ref</th>\n",
       "      <th>Seq_Obs</th>\n",
       "      <th>Seq_Unobs</th>\n",
       "      <th>Check_RefMatch</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr4</td>\n",
       "      <td>74487576</td>\n",
       "      <td>74488141</td>\n",
       "      <td>chr4:74487576-74488141</td>\n",
       "      <td>chr4:74487586:A:G:C</td>\n",
       "      <td>74487586</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>0.964984</td>\n",
       "      <td>35</td>\n",
       "      <td>chr4:74487551-74487622</td>\n",
       "      <td>71</td>\n",
       "      <td>ACATGTGTGTATGCTTGTCTATATACACTGTATGAAGCAAGAGGAG...</td>\n",
       "      <td>ACATGTGTGTATGCTTGTCTATATACACTGTATGAGGCAAGAGGAG...</td>\n",
       "      <td>ACATGTGTGTATGCTTGTCTATATACACTGTATGACGCAAGAGGAG...</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr9</td>\n",
       "      <td>135685425</td>\n",
       "      <td>135685942</td>\n",
       "      <td>chr9:135685425-135685942</td>\n",
       "      <td>chr9:135685432:G:G:C</td>\n",
       "      <td>135685432</td>\n",
       "      <td>G</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>0.945423</td>\n",
       "      <td>35</td>\n",
       "      <td>chr9:135685397-135685468</td>\n",
       "      <td>71</td>\n",
       "      <td>CCGGGGACACATCCCCCAGGAAACCTCTGGGCTGAGGCAACACCAA...</td>\n",
       "      <td>CCGGGGACACATCCCCCAGGAAACCTCTGGGCTGAGGCAACACCAA...</td>\n",
       "      <td>CCGGGGACACATCCCCCAGGAAACCTCTGGGCTGACGCAACACCAA...</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>173251481</td>\n",
       "      <td>173251541</td>\n",
       "      <td>chr1:173251481-173251541</td>\n",
       "      <td>chr1:173251490:G:G:C</td>\n",
       "      <td>173251490</td>\n",
       "      <td>G</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>0.910127</td>\n",
       "      <td>35</td>\n",
       "      <td>chr1:173251455-173251526</td>\n",
       "      <td>71</td>\n",
       "      <td>ACATAGTACTCAATACCTATTTGGTGAATAGATGAGGCAATCTCAC...</td>\n",
       "      <td>ACATAGTACTCAATACCTATTTGGTGAATAGATGAGGCAATCTCAC...</td>\n",
       "      <td>ACATAGTACTCAATACCTATTTGGTGAATAGATGACGCAATCTCAC...</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>147523084</td>\n",
       "      <td>147523503</td>\n",
       "      <td>chr1:147523084-147523503</td>\n",
       "      <td>chr1:147523391:G:G:C</td>\n",
       "      <td>147523391</td>\n",
       "      <td>G</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>0.908800</td>\n",
       "      <td>35</td>\n",
       "      <td>chr1:147523356-147523427</td>\n",
       "      <td>71</td>\n",
       "      <td>ATTTTTAAAAGTTCAGGTAGTTTAAAGATGGCTGAGGCAATGCAAT...</td>\n",
       "      <td>ATTTTTAAAAGTTCAGGTAGTTTAAAGATGGCTGAGGCAATGCAAT...</td>\n",
       "      <td>ATTTTTAAAAGTTCAGGTAGTTTAAAGATGGCTGACGCAATGCAAT...</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr18</td>\n",
       "      <td>54580694</td>\n",
       "      <td>54581420</td>\n",
       "      <td>chr18:54580694-54581420</td>\n",
       "      <td>chr18:54580749:A:G:T</td>\n",
       "      <td>54580749</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>0.862296</td>\n",
       "      <td>35</td>\n",
       "      <td>chr18:54580714-54580785</td>\n",
       "      <td>71</td>\n",
       "      <td>TGTTTAGCTAACAGAAATACTTCAGCAGAGCCTGAAGCAATGGCCA...</td>\n",
       "      <td>TGTTTAGCTAACAGAAATACTTCAGCAGAGCCTGAGGCAATGGCCA...</td>\n",
       "      <td>TGTTTAGCTAACAGAAATACTTCAGCAGAGCCTGATGCAATGGCCA...</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   Chrom  ChromStart   ChromEnd                    Region  \\\n",
       "0   chr4    74487576   74488141    chr4:74487576-74488141   \n",
       "1   chr9   135685425  135685942  chr9:135685425-135685942   \n",
       "2   chr1   173251481  173251541  chr1:173251481-173251541   \n",
       "3   chr1   147523084  147523503  chr1:147523084-147523503   \n",
       "4  chr18    54580694   54581420   chr18:54580694-54581420   \n",
       "\n",
       "             Variant_ID       Pos0 Ref Obs Unobs     Delta  Flank  \\\n",
       "0   chr4:74487586:A:G:C   74487586   A   G     C  0.964984     35   \n",
       "1  chr9:135685432:G:G:C  135685432   G   G     C  0.945423     35   \n",
       "2  chr1:173251490:G:G:C  173251490   G   G     C  0.910127     35   \n",
       "3  chr1:147523391:G:G:C  147523391   G   G     C  0.908800     35   \n",
       "4  chr18:54580749:A:G:T   54580749   A   G     T  0.862296     35   \n",
       "\n",
       "                   Interval  Seq_Length  \\\n",
       "0    chr4:74487551-74487622          71   \n",
       "1  chr9:135685397-135685468          71   \n",
       "2  chr1:173251455-173251526          71   \n",
       "3  chr1:147523356-147523427          71   \n",
       "4   chr18:54580714-54580785          71   \n",
       "\n",
       "                                             Seq_Ref  \\\n",
       "0  ACATGTGTGTATGCTTGTCTATATACACTGTATGAAGCAAGAGGAG...   \n",
       "1  CCGGGGACACATCCCCCAGGAAACCTCTGGGCTGAGGCAACACCAA...   \n",
       "2  ACATAGTACTCAATACCTATTTGGTGAATAGATGAGGCAATCTCAC...   \n",
       "3  ATTTTTAAAAGTTCAGGTAGTTTAAAGATGGCTGAGGCAATGCAAT...   \n",
       "4  TGTTTAGCTAACAGAAATACTTCAGCAGAGCCTGAAGCAATGGCCA...   \n",
       "\n",
       "                                             Seq_Obs  \\\n",
       "0  ACATGTGTGTATGCTTGTCTATATACACTGTATGAGGCAAGAGGAG...   \n",
       "1  CCGGGGACACATCCCCCAGGAAACCTCTGGGCTGAGGCAACACCAA...   \n",
       "2  ACATAGTACTCAATACCTATTTGGTGAATAGATGAGGCAATCTCAC...   \n",
       "3  ATTTTTAAAAGTTCAGGTAGTTTAAAGATGGCTGAGGCAATGCAAT...   \n",
       "4  TGTTTAGCTAACAGAAATACTTCAGCAGAGCCTGAGGCAATGGCCA...   \n",
       "\n",
       "                                           Seq_Unobs  Check_RefMatch  \n",
       "0  ACATGTGTGTATGCTTGTCTATATACACTGTATGACGCAAGAGGAG...            True  \n",
       "1  CCGGGGACACATCCCCCAGGAAACCTCTGGGCTGACGCAACACCAA...            True  \n",
       "2  ACATAGTACTCAATACCTATTTGGTGAATAGATGACGCAATCTCAC...            True  \n",
       "3  ATTTTTAAAAGTTCAGGTAGTTTAAAGATGGCTGACGCAATGCAAT...            True  \n",
       "4  TGTTTAGCTAACAGAAATACTTCAGCAGAGCCTGATGCAATGGCCA...            True  "
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "%%time\n",
    "\n",
    "### init\n",
    "dat = dat_variant_arrange.copy()\n",
    "\n",
    "### add sequences\n",
    "fun = partial(get_variant_interval_series, num_interval_flank=NUM_INTERVAL_FLANK)\n",
    "lst = [\"Flank\", \"Interval\", \"Seq_Length\", \"Seq_Ref\", \"Seq_Obs\", \"Seq_Unobs\", \"Check_RefMatch\"]\n",
    "dat[lst] = dat.apply(fun, axis=1)\n",
    "\n",
    "### assign and show\n",
    "dat_variant_withseq_flank35 = dat\n",
    "print(dat.shape)\n",
    "dat.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "d8c1d36b-21a1-41c3-ba20-47256db36f4c",
   "metadata": {},
   "source": [
    "**Export results**"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "8060b49b-4b00-471c-afd3-1a5353307672",
   "metadata": {},
   "outputs": [],
   "source": [
    "### set file directory for table with sequences\n",
    "txt_fdiry = os.path.join(FD_RES, \"analysis_variant_motif_richard\")\n",
    "txt_fname = \"variant_closed_gof_bluestarr_withseq_flank35.tsv.gz\"\n",
    "txt_fpath = os.path.join(txt_fdiry, txt_fname)\n",
    "\n",
    "### write table with sequences\n",
    "dat_variant_withseq_flank35.to_csv(txt_fpath, sep=\"\\t\", index=False, compression=\"gzip\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "aeb24a5d-2b59-461a-a6d6-43a7928e8637",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "4e4811aa-8dcd-44cc-a829-2f3a3729206e",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "4c825f82-f561-4a1e-be47-ed3fd7fdee82",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "id": "9620d558-887a-45ca-9fc6-5b7eece59335",
   "metadata": {},
   "source": [
    "## Test function"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "e51772c9-6271-4c60-a457-e02ecf4c9e73",
   "metadata": {},
   "outputs": [],
   "source": [
    "def get_variant_interval(\n",
    "    txt_chrom_name, \n",
    "    txt_chrom_pos0, \n",
    "    txt_allele_ref, \n",
    "    txt_allele_obs, \n",
    "    txt_allele_unobs, \n",
    "    num_interval_flank = 35\n",
    "):\n",
    "    \"\"\"\n",
    "    Get reference, observed, and unobserved allele windows around a variant.\n",
    "\n",
    "    Parameters\n",
    "    ----------\n",
    "    txt_chrom_name : str\n",
    "        Chromosome name (e.g., \"chr4\").\n",
    "    txt_chrom_pos0 : int\n",
    "        Variant position (0-based coordinate).\n",
    "    txt_allele_ref : str\n",
    "        Reference allele from the table.\n",
    "    txt_allele_obs : str\n",
    "        Observed alternate allele.\n",
    "    txt_allele_unobs : str\n",
    "        Unobserved alternate allele.\n",
    "    num_interval_flank : int, default=35\n",
    "        Number of bases to include on each side of the variant.\n",
    "\n",
    "    Returns\n",
    "    -------\n",
    "    tuple of str\n",
    "        (seq_ref, seq_obs, seq_unobs) sequences of length 2*flank+1.\n",
    "    \"\"\"\n",
    "    ### extract reference window from FASTA\n",
    "    txt_seq_ref = fasta.fetch(\n",
    "        txt_chrom_name,\n",
    "        txt_chrom_pos0 - num_interval_flank,\n",
    "        txt_chrom_pos0 + num_interval_flank + 1\n",
    "    )\n",
    "\n",
    "    ### sanity check central base\n",
    "    if txt_seq_ref[num_interval_flank].upper() != txt_allele_ref.upper():\n",
    "        raise ValueError(\n",
    "            f\"Reference mismatch at {txt_chrom_name}:{txt_chrom_pos0} \"\n",
    "            f\"(expected {txt_allele_ref}, got {seq_ref[num_interval_flank]})\"\n",
    "        )\n",
    "\n",
    "    ### build alt windows\n",
    "    txt_seq_obs   = txt_seq_ref[:num_interval_flank] + txt_allele_obs   + txt_seq_ref[num_interval_flank+1:]\n",
    "    txt_seq_unobs = txt_seq_ref[:num_interval_flank] + txt_allele_unobs + txt_seq_ref[num_interval_flank+1:]\n",
    "\n",
    "    return txt_seq_ref, txt_seq_obs, txt_seq_unobs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "id": "68dde05b-0f0e-4cab-9729-6012dacce120",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ref   : acatgtgtgtatgcttgtctatataCACTGTATGAAGCAAGAGGAGAAAATACTCATGAATTCTAGGAACT\n",
      "obs   : acatgtgtgtatgcttgtctatataCACTGTATGAGGCAAGAGGAGAAAATACTCATGAATTCTAGGAACT\n",
      "unobs : acatgtgtgtatgcttgtctatataCACTGTATGACGCAAGAGGAGAAAATACTCATGAATTCTAGGAACT\n"
     ]
    }
   ],
   "source": [
    "row = dat_variant_import.iloc[0]\n",
    "\n",
    "txt_chrom_name = row[\"region\"].split(\":\")[0]\n",
    "txt_chrom_pos0 = row[\"pos0\"]\n",
    "txt_allele_ref = row[\"ref\"]\n",
    "txt_allele_obs = row[\"obs\"]\n",
    "txt_allele_unobs = row[\"unobs\"]\n",
    "\n",
    "txt_seq_ref, txt_seq_obs, txt_seq_unobs = get_variant_interval(\n",
    "    txt_chrom_name, txt_chrom_pos0, \n",
    "    txt_allele_ref, txt_allele_obs, txt_allele_unobs,\n",
    "    num_interval_flank = NUM_INTERVAL_FLANK\n",
    ")\n",
    "\n",
    "print(\"ref   :\", txt_seq_ref)\n",
    "print(\"obs   :\", txt_seq_obs)\n",
    "print(\"unobs :\", txt_seq_unobs)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "id": "a5b9d678-0de7-4b5c-923d-dd18b074c6cb",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(10000, 9)\n",
      "CPU times: user 2.14 s, sys: 225 ms, total: 2.36 s\n",
      "Wall time: 2.38 s\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>region</th>\n",
       "      <th>pos0</th>\n",
       "      <th>delta</th>\n",
       "      <th>ref</th>\n",
       "      <th>obs</th>\n",
       "      <th>unobs</th>\n",
       "      <th>seq_ref</th>\n",
       "      <th>seq_obs</th>\n",
       "      <th>seq_unobs</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>771970</th>\n",
       "      <td>chr4:183063682-183063779</td>\n",
       "      <td>183063712</td>\n",
       "      <td>0.006892</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>AAAGAAATACAGAATTTTTCCCATGGTCACTCTTCTGAGGCACAGT...</td>\n",
       "      <td>AAAGAAATACAGAATTTTTCCCATGGTCACTCTTCCGAGGCACAGT...</td>\n",
       "      <td>AAAGAAATACAGAATTTTTCCCATGGTCACTCTTCAGAGGCACAGT...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>256899</th>\n",
       "      <td>chr7:115860432-115860716</td>\n",
       "      <td>115860564</td>\n",
       "      <td>0.015483</td>\n",
       "      <td>T</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>GAATAACACAGAGCTTTGAGTTCACCCAAACCCATTGTGATCAAGG...</td>\n",
       "      <td>GAATAACACAGAGCTTTGAGTTCACCCAAACCCATTGTGATCAAGG...</td>\n",
       "      <td>GAATAACACAGAGCTTTGAGTTCACCCAAACCCATCGTGATCAAGG...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1618901</th>\n",
       "      <td>chr9:116092981-116093246</td>\n",
       "      <td>116093025</td>\n",
       "      <td>0.001214</td>\n",
       "      <td>T</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>AAGCAGTTCTCTAAGACTCTAGTTAGTAAATTGAATATTCATTTAA...</td>\n",
       "      <td>AAGCAGTTCTCTAAGACTCTAGTTAGTAAATTGAATATTCATTTAA...</td>\n",
       "      <td>AAGCAGTTCTCTAAGACTCTAGTTAGTAAATTGAAAATTCATTTAA...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>149278</th>\n",
       "      <td>chr1:152521537-152522216</td>\n",
       "      <td>152521539</td>\n",
       "      <td>0.020028</td>\n",
       "      <td>G</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>ttaatttaaagtagacaacagcaagtgcaaaggTTGGAAAGGTGAT...</td>\n",
       "      <td>ttaatttaaagtagacaacagcaagtgcaaaggTTGGAAAGGTGAT...</td>\n",
       "      <td>ttaatttaaagtagacaacagcaagtgcaaaggTTTGAAAGGTGAT...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>201620</th>\n",
       "      <td>chr7:34886252-34886354</td>\n",
       "      <td>34886288</td>\n",
       "      <td>0.017478</td>\n",
       "      <td>A</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>GCTATTATCTAGTTCCTCATGATTAATGCTGTGTTAGGACTCCAGA...</td>\n",
       "      <td>GCTATTATCTAGTTCCTCATGATTAATGCTGTGTTAGGACTCCAGA...</td>\n",
       "      <td>GCTATTATCTAGTTCCTCATGATTAATGCTGTGTTTGGACTCCAGA...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                           region       pos0     delta ref obs unobs  \\\n",
       "771970   chr4:183063682-183063779  183063712  0.006892   T   C     A   \n",
       "256899   chr7:115860432-115860716  115860564  0.015483   T   T     C   \n",
       "1618901  chr9:116092981-116093246  116093025  0.001214   T   T     A   \n",
       "149278   chr1:152521537-152522216  152521539  0.020028   G   G     T   \n",
       "201620     chr7:34886252-34886354   34886288  0.017478   A   A     T   \n",
       "\n",
       "                                                   seq_ref  \\\n",
       "771970   AAAGAAATACAGAATTTTTCCCATGGTCACTCTTCTGAGGCACAGT...   \n",
       "256899   GAATAACACAGAGCTTTGAGTTCACCCAAACCCATTGTGATCAAGG...   \n",
       "1618901  AAGCAGTTCTCTAAGACTCTAGTTAGTAAATTGAATATTCATTTAA...   \n",
       "149278   ttaatttaaagtagacaacagcaagtgcaaaggTTGGAAAGGTGAT...   \n",
       "201620   GCTATTATCTAGTTCCTCATGATTAATGCTGTGTTAGGACTCCAGA...   \n",
       "\n",
       "                                                   seq_obs  \\\n",
       "771970   AAAGAAATACAGAATTTTTCCCATGGTCACTCTTCCGAGGCACAGT...   \n",
       "256899   GAATAACACAGAGCTTTGAGTTCACCCAAACCCATTGTGATCAAGG...   \n",
       "1618901  AAGCAGTTCTCTAAGACTCTAGTTAGTAAATTGAATATTCATTTAA...   \n",
       "149278   ttaatttaaagtagacaacagcaagtgcaaaggTTGGAAAGGTGAT...   \n",
       "201620   GCTATTATCTAGTTCCTCATGATTAATGCTGTGTTAGGACTCCAGA...   \n",
       "\n",
       "                                                 seq_unobs  \n",
       "771970   AAAGAAATACAGAATTTTTCCCATGGTCACTCTTCAGAGGCACAGT...  \n",
       "256899   GAATAACACAGAGCTTTGAGTTCACCCAAACCCATCGTGATCAAGG...  \n",
       "1618901  AAGCAGTTCTCTAAGACTCTAGTTAGTAAATTGAAAATTCATTTAA...  \n",
       "149278   ttaatttaaagtagacaacagcaagtgcaaaggTTTGAAAGGTGAT...  \n",
       "201620   GCTATTATCTAGTTCCTCATGATTAATGCTGTGTTTGGACTCCAGA...  "
      ]
     },
     "execution_count": 21,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "%%time\n",
    "\n",
    "dat = dat_variant_import.copy()\n",
    "dat = dat.sample(10000, random_state=1)\n",
    "\n",
    "\n",
    "fun = partial(get_variant_interval_series, num_interval_flank=NUM_INTERVAL_FLANK)\n",
    "dat[[\"interval\", \"seq_ref\", \"seq_obs\", \"seq_unobs\"]] = dat.apply(\n",
    "    fun, axis=1\n",
    ")\n",
    "\n",
    "dat_variant_arrange = dat\n",
    "print(dat.shape)\n",
    "dat.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "836545c2-93f4-46ba-bdce-03015aca3170",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "c35786e9-2407-4f7d-abfe-e08dcde489c8",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.11.6"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
